### DISCUSSION TITLE
Thinking to use Grant JS for OAuth and managing session with iron-session

### DISCUSSION THREAD

**Discussion by Syed-Sheharyar** [vvo/iron-session#655] [Q&A]
I am starting a new project and there is again authentication part that needs to be covered. I am thinking about using [Grant JS](https://github.com/simov/grant) and iron-session for session management. Now, Grant JS provides more than 200+ OAuth providers and we can set up complete OAuth by just using one config file which surprises me. 

Still, no one talks about it too much and there is not a single article on using this with any technology like express js or building something and using this as an OAuth solution. However, few articles recommend Grant JS in the top 5 best authentication solutions for Node JS. Grant JS cannot be used alone it requires some kind of session management which I think iron-session does a good job.

So, Now I want to know what you think about Grant JS @vvo and when should one use it from the readme and a few articles from the author on Medium it looks like the fastest way to add authentication using OAuth. Then, manage the session using iron-session. The real problem is that there is not a single example of using it with Next JS. For now, I think I should be using `grant().vercel()` but then I don't want hosting to be limited to Vercel only. Don't know if I can host it on a regular VPS and auth works even after using `grant().vercel()`.

**Comment by vvo** [comment]
@Syed-Sheharyar Grant JS seems like a nice way to easily integration a "login with Slack/Google/whatever" solution and yes then you can store the tokens in iron-session. Is there anything else you'd like to know?

**Reply by Syed-Sheharyar** [reply]
https://www.reddit.com/r/nextjs/comments/18aqmj0/anybody_can_help_with_authentication/
https://github.com/simov/grant/issues/297

Can you help me with integrating Grant and iron-session in Next JS? Read the Reddit Post for full insight. @vvo

**Reply by vvo** [reply]
Sorry, I have no idea how to do this

**Comment by subhasish-smiles** [comment]
This I needed as well
